ovs/lib/storagedriver.py

```diff
@@ -165,71 +187,95 @@ def manual_voldrv_arakoon_checkup():
             new_service.save()
             return new_service
 
-        cluster_name = 'voldrv'
+        claimed_clusters = ArakoonInstaller.get_arakoon_metadata_by_cluster_type(cluster_type=ServiceType.ARAKOON_CLUSTER_TYPES.SD, in_use=True)
+        if len(claimed_clusters) > 1:
+            raise ValueError('Found multiple clusters of type "{0}" which have already been claimed'.format(ServiceType.ARAKOON_CLUSTER_TYPES.SD))
+
         service_name = 'arakoon-voldrv'
-        service_type = ServiceTypeList.get_by_name('Arakoon')
-        current_services = []
+        service_type = ServiceTypeList.get_by_name(ServiceType.SERVICE_TYPES.ARAKOON)
+
         current_ips = []
+        current_services = []
         for service in service_type.services:
             if service.name == service_name:
                 current_services.append(service)
-                current_ips.append(service.storagerouter.ip)
+                if service.is_internal is True:
+                    current_ips.append(service.storagerouter.ip)
+
         all_sr_ips = [storagerouter.ip for storagerouter in StorageRouterList.get_slaves()]
         available_storagerouters = {}
         for storagerouter in StorageRouterList.get_masters():
             storagerouter.invalidate_dynamics(['partition_config'])
             if len(storagerouter.partition_config[DiskPartition.ROLES.DB]) > 0:
                 available_storagerouters[storagerouter] = DiskPartition(storagerouter.partition_config[DiskPartition.ROLES.DB][0])
             all_sr_ips.append(storagerouter.ip)
-        if create_cluster is True and len(current_services) == 0 and len(available_storagerouters) > 0:
-            storagerouter, partition = available_storagerouters.items()[0]
-            result = ArakoonInstaller.create_cluster(cluster_name=cluster_name,
-                                                     ip=storagerouter.ip,
-                                                     base_dir=partition.folder)
-            current_services.append(add_service(storagerouter, result))
-            ArakoonInstaller.restart_cluster_add(cluster_name, current_ips, storagerouter.ip)
-            current_ips.append(storagerouter.ip)
-            StorageDriverController._configure_arakoon_to_volumedriver()
-
-        if 0 < len(current_services) < len(available_storagerouters):
+
+        if create_cluster is True and len(claimed_clusters) == 0 and len(current_services) == 0:  # Create new cluster
+            metadata = ArakoonInstaller.get_arakoon_metadata_by_cluster_type(cluster_type=ServiceType.ARAKOON_CLUSTER_TYPES.SD, in_use=False)
+            if len(metadata) == 0:  # No externally managed cluster found, we create 1 ourselves
+                if not available_storagerouters:
+                    raise RuntimeError('Could not find any Storage Router with a DB role')
+
+                storagerouter, partition = available_storagerouters.items()[0]
+                cluster_name = 'voldrv'
```

Comment:
You should make sure that if another cluster would be called 'voldrv' that this code can either generate another name and continue, or that the create_cluster raises that such cluster already exists. Now it might just silently mess up configurations.